static/src/javascripts/projects/common/modules/experiments/ab.js

```diff
@@ -157,6 +161,60 @@ define([
         return tag.join(',');
     }
 
+    function getAbLoggableObject() {
+        var abLogObject = {};
+
+        try {
+            forEach(getActiveTests(), function (test) {
+                if (isParticipating(test) && testCanBeRun(test)) {
+                    var variant = getTestVariantId(test.id);
+                    if (variant && variant !== 'notintest') {
+                        merge(abLogObject, createAbObject(test.id, variant, 'false'));
+                    }
+                }
+            });
+            forEach(getServerSideTests(), function (testName) {
+                merge(abLogObject, createAbObject('ab' + testName, 'inTest', 'false'));
+            });
+        } catch (error) {
+            // Encountering an error should invalidate the logging process.
+            abLogObject = {};
+            reportError(error, false);
+        }
+
+        return abLogObject;
+    }
+
+    function createAbObject(id, variantName, complete) {
+        var abData = {};
```

Comment:
Couldn't you get rid of this object and just return its content? This would spare you the trouble of merging all the time.